app/controllers/RegisteredAddressForEoriController.scala

```diff
@@ -39,36 +39,35 @@ class RegisteredAddressForEoriController @Inject()(appConfig: FrontendAppConfig,
                                       navigator: Navigator,
                                       identify: IdentifierAction,
                                       getData: DataRetrievalAction,
+                                      requireData: DataRequiredAction,
                                       formProvider: RegisteredAddressForEoriFormProvider
                                       ) extends FrontendController with I18nSupport {
 
   val form = formProvider()
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) {
-    implicit request =>
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) { implicit request =>
 
-      val preparedForm = request.userAnswers.flatMap(_.get(RegisteredAddressForEoriPage)) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
+    val preparedForm = request.userAnswers.get(RegisteredAddressForEoriPage) match {
+      case None => form
+      case Some(value) => form.fill(value)
+    }
 
-      Ok(registeredAddressForEori(appConfig, preparedForm, mode))
+    Ok(registeredAddressForEori(appConfig, preparedForm, mode))
   }
 
-  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData).async {
-    implicit request =>
+  def onSubmit(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
 
-      form.bindFromRequest().fold(
-        (formWithErrors: Form[_]) =>
-          Future.successful(BadRequest(registeredAddressForEori(appConfig, formWithErrors, mode))),
-        value => {
-          val updatedAnswers = request.userAnswers.getOrElse(UserAnswers(request.internalId)).set(RegisteredAddressForEoriPage, value)
```

Comment:
How come this was removed?

Reply:
It looks to me like it's protecting against `request.userAnswers` being `None` which I imagine will happen on the first time you go through the journey. Need to double check this still works

Reply:
Just give it a go in the UI, if it works then happy days